frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber/EditNumber.tsx

```diff
@@ -25,44 +29,112 @@ type EditNumberProps = EditFieldProps<NumberFieldBase>
 
 const EDIT_NUMBER_FIELD_KEYS = ['title', 'description', 'required'] as const
 
+// As we want to keep the values in the shared type simple,
+// we create a separate enum for frontend options and transform them as needed
+enum NumberSelectedValidationInputs {
+  Length = 'Number of characters allowed',
+  Range = 'Range of values allowed',
+}
+
 type EditNumberInputs = Pick<
   NumberFieldBase,
   typeof EDIT_NUMBER_FIELD_KEYS[number]
 > & {
   ValidationOptions: {
-    selectedValidation: NumberSelectedValidation | ''
-    customVal: number | ''
+    selectedValidation: NumberSelectedValidationInputs | ''
+    LengthValidationOptions: {
+      customVal: number | ''
+      selectedLengthValidation: NumberSelectedLengthValidation
+    }
+    RangeValidationOptions: {
+      customMin: number | ''
+      customMax: number | ''
+    }
   }
 }
 
 const transformNumberFieldToEditForm = (
   field: NumberFieldBase,
 ): EditNumberInputs => {
-  const nextValidationOptions = {
-    selectedValidation:
-      field.ValidationOptions.selectedValidation || ('' as const),
+  const {
+    selectedValidation,
+    LengthValidationOptions,
+    RangeValidationOptions,
+  } = field.ValidationOptions
+
+  const nextSelectedValidation =
+    selectedValidation === NumberSelectedValidation.Length
+      ? NumberSelectedValidationInputs.Length
+      : selectedValidation === NumberSelectedValidation.Range
+      ? NumberSelectedValidationInputs.Range
+      : ('' as const)
+
+  const nextLengthValidationOptions = {
+    selectedLengthValidation:
+      LengthValidationOptions.selectedLengthValidation ||
+      NumberSelectedLengthValidation.Min,
```

Comment:
Can we default to an empty string here initially? We should then have validation on the dropdown to ensure it is filled upon save.

Reply:
Just to confirm, do we also want it to be empty whenever the validation is cleared?

Reply:
I think that may not be necessary. If a user 1) selects that validation, 2) enters their customization, 3) then they clear the selected validation, but 4) re-select it later, we should default to their original customization at step 2. So I don't think we need to add instrumentation to clear the input.

Reply:
Could I confirm the desired behaviour for each input when clearing selected validation? We currently have the following inputs:
1. Selected length
2. Length `customVal`
3. `rangeMin`
4. `rangeMax`

Originally, `customVal` was cleared along with the selected validation type, and I was wondering if we would like to keep this behaviour for all the inputs.

Reply:
Hmm... Ok, thinking about it again, i think this makes sense. Let's retain the behavior of clearing all the 4 inputs you pointed out, when the selected validation is cleared. Thanks!